src/Symfony/Component/String/Resources/functions.php

```diff
@@ -11,22 +11,28 @@
 
 namespace Symfony\Component\String;
 
-function u(?string $string = ''): UnicodeString
-{
-    return new UnicodeString($string ?? '');
+if (!\function_exists('Symfony\Component\String\u')) {
```

Comment:
Shall we use `__NAMESPACE__` here, to guard ourselves against typos?

```suggestion
if (!\function_exists(__NAMESPACE__.'\u')) {
```

Reply:
or just `if (!\function_exists(u::class)) {`

Reply:
maybe we should have a single guard clause?
```php
if (!\function_exists(u::class)) {
    return;
}
```

Reply:
We used a single gard in Guzzle. That worked great until we added a new function. Now people installed an old version and a new version at the same time and they experienced even more issues.

Reply:
I don't like this at all, but you're right, we don't have a better option, unfortunately...

Reply:
Yeah. The alternative would be to define a `Util` class with static methods. That is not better...